proposed/http-message.md

```diff
@@ -490,6 +501,58 @@ namespace Psr\Http\Message;
 interface RequestInterface extends MessageInterface
 {
     /**
+     * Extends MessageInterface::getHeaders() to provide request-specific
+     * behavior.
+     *
+     * Retrieves all message headers.
+     *
+     * This method acts exactly like MessageInterface::getHeaders(), with one
+     * behavioral change: if the Host header has not been previously set, the
+     * method MUST attempt to pull the host segment of the composed URI, if
```

Comment:
Disagree, :-1: there's no significance to the order in the array.

It's up to a client to correctly produce the http request and put the host first.